### PR TITLE
Updated ADO pipeline names & update GitHub workflow conditions to not skip over tests

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Updated workflow conditions to not skip over tests
- Aligned Azure DevOps pipeline files with GitHub composite actions names

Pipeline reference
- Expected continous run when first pester task fails: https://github.com/Azure/ResourceModules/runs/4920105211?check_suite_focus=true (Note: The API tests ran now after all)
- Exected cancellation when job is cancled: https://github.com/Azure/ResourceModules/actions/runs/1739266888  (Note: This test is supposed to ensure cancellations still work)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: utilities/tools/AzureDevOps/src/devOpsPipelineTemplate.yml [hunk 1]
```diff
@@ -41,12 +41,12 @@ stages:
   - stage: Validation
     displayName: Pester tests
     jobs:
-      - template: /.azuredevops/pipelineTemplates/module.jobs.validate.yml
+      - template: /.azuredevops/pipelineTemplates/jobs.validateModulePester.yml
```
Comment (eriqua) [inline utilities/tools/AzureDevOps/src/devOpsPipelineTemplate.yml:44]: Is this template used or intended just as a reference?

Comment (MrMCake) [reply utilities/tools/AzureDevOps/src/devOpsPipelineTemplate.yml:44]: It's a template for a tool that I wrote back in the day to generate the ADO pipelines from the GitHub counterpart. Thinking of it we may want to question whether we want to keep it or not. What do you think?

Comment (MrMCake) [reply utilities/tools/AzureDevOps/src/devOpsPipelineTemplate.yml:44]: I guess it's more to maintain and now we have 'permanent' ADO pipelines

Comment (MrMCake) [reply utilities/tools/AzureDevOps/src/devOpsPipelineTemplate.yml:44]: I'd almost be tempted to remove the logic as part of this PR

Comment (eriqua) [reply utilities/tools/AzureDevOps/src/devOpsPipelineTemplate.yml:44]: If not used I'd opt for removing it

Comment (MrMCake) [reply utilities/tools/AzureDevOps/src/devOpsPipelineTemplate.yml]: Removed

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**